### PR TITLE
@acjay => React-Jade cleanup 

### PR DESCRIPTION
**Pull request by damassi**
Slight cleanup via HOC so users don't have to compile jade templates before passing into `.jsx` route; also file renaming for clarity and stub of `/auctions2`. 

**Route:**

```javascript
export function index (req, res, next) {
  res.render('index.jsx', {
    basePath: res.app.get('views'),
    locals: {
      ...res.locals,
      myActiveBids: ['a', 'b', 'c']
    }),
    templates: {
      meta: 'meta.jade',
      myActiveBids: 'my_active_bids.jade'
    }
  })
}
```

**templates/index.jsx:**

```javascript
import { withLayout } from 'desktop/components/react/main_layout'

function Index (props) {
  const {
    Layout: { Header, Body, Footer },
    Templates: { Meta, MyActiveBids }
  } = props

  return (
    <div>
      <Header>
        <Meta />
      </Header>
      <Body>
        <MyActiveBids /> 
      </Body>
      <Footer />
    </div>
  )
}

export default withLayout(Index)
```

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please assign someone to your PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: desktop/apps/auctions2/queries/index.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+export function bidderPositions () {
+  return `{
+    me {
+      bidder_positions {
+        id
+      }
+    }
+  }`
```
Comment (damassi) [inline desktop/apps/auctions2/queries/index.js]: Just stubs for `/auction2`; pls ignore

File: desktop/apps/auctions2/templates/index.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  return (
+    <div>
+      <Header>
+        <Meta />
+      </Header>
+      <Body>
+        hello!
+      </Body>
+      <Footer />
+    </div>
```
Comment (acjay) [inline desktop/apps/auctions2/templates/index.jsx]: I still wonder what we're really expressing here. It doesn't seem like we'll ever really want to rearrange the order of the `Layout`'s components. So, why not make that framing implicit? Something like:

```
import PropTypes from 'prop-types'
import React from 'react'
import { withLayout } from 'desktop/components/react/main_layout'

function Index () {
  return (
    <div>hello!</div>
  )
}

export default withLayout(Index)
```

So `withLayout` would effectively inject its component into the body block. Or I guess it would probably be less of an `HOC` than a renderer, since the outcome would be a fully rendered layout, instead of a component. I confirmed locally that it's possible to put raw HTML in a Pug `block`.

Then, to the extent that we need to be able to hook into those other sections, perhaps that could be done as additional arguments to `withLayout`. So you'd have `withLayout(Index, { header })` if you needed to add something to the header.

Or, if we need a lot of control over ways to manipulate the subsections, `withLayout` could become basically a constructor for a [fluent interface](https://en.wikipedia.org/wiki/Fluent_interface#JavaScript) that offers the ability to attach or overwrite the different sections, with either React components or raw HTML. So you could have operations like `replaceFooterWithComponent`, `appendRawHtmlToHeader`, or `appendMeta`, and terminate with a `renderToString()`.

Comment (damassi) [inline desktop/apps/auctions2/templates/index.jsx]: If you dig into subapps in force, each one expresses its own header, footer, etc -- there isn't one generic wrapper that we reuse all over. Each has different attributes and plugs in, so to speak.

As this pattern is still evolving (and will become clearer) I'd like to keep going in the direction I'm going without any new features; when we get into `/auction2` territory hopefully it will make more sense. Right now I feel this is a simpler approach since we don't yet need a lot of control except in places where `block` portals occur (the point of this all), and the header, body, footer approach gets it done quick and easy without much overhead.

Comment (acjay) [inline desktop/apps/auctions2/templates/index.jsx]: Hmm, maybe I'm not fully understanding. I think what I'm confused about is why this template needs to mention `Header` and `Meta`.

Comment (acjay) [inline desktop/apps/auctions2/templates/index.jsx]: Like, it seems to me that a given app is either going to use the old `block` way to insert content into its subsections or it will want to manage that content using a React component. That's where I'm coming from here -- if it's the old way, then why have to mention these `Layout` subsections at all? If it's the new way, then rather than trying to control those subsections in one super component that is responsible for composing the layout with the main content and custom subsections, does it make more sense to just provide the main content and custom subsections as separate pieces of data, to be composed into the layout by one central piece of functionality. 

So basically, I'm suggesting separating the concerns of full-page layout composition and app specific content generation by pulling those layout concerns out of the app-specific code, and just having the app-specific code supply the layout data, not the layout logic.

Comment (damassi) [inline desktop/apps/auctions2/templates/index.jsx]: Ahh sorry, let me explain, details are definitely missing :) 

So pug has this Inheritance feature https://pugjs.org/language/inheritance.html that allows you to break apart templates into "zones" and inject that content into "blocks", effectively acting as a portal. 

```
block foo

block foo
  .bar 
    | this content will be injected into the top named block
```
The way force was designed is that the main layout has `block header`, `block `body` and `block footer` scattered about. These are the portals in which sub-apps, when mounted, inject their content into. Each app usually has an `index.jade` that imports `main_layout`, thereby setting it up to receive content imported from a local `header.jade` and `footer.jade` , etc. each split based upon their own tagged `block header` sections. 

Right now this whole process is a black box and _given the architecture that already exists_ there is no way to use anything but jade to get content into those portals. Hence the need to split them up and define a system that renders the content into three _generic_ sections -- header, body, and footer -- which then accepts, as `children` subapp template code. 

I should have used something other than `meta` as an example, but that simply refers to the filename, and could be anything. What it represents is simply a reactified block of jade code.

Some more docs would have helped:

```javascript
res.view('index.jsx', {
  // Anything under `templates` is transformed into a react component 
  // and can be used wherever
  templates: {
     myTemplate: 'my_template.jade' // => <MyTemplate />
  }
})

// templates/index.jsx

function Index (props) {
  const {
    Layout: { Header, Body, Footer },
    Templates: { MyTemplate }
  } = props

  return (
    <div>
      <Header>
        <MyTemplate />
      </Header>
      <Body>
        ... 
      </Body>
      <Footer />
    </div>
  )
}
```

Since it is typical to have jade fragments that go in the header or the footer or the body -- delineated by `block` portals -- those three sections have to be returned and available to accept content. 

Does this make sense? Its really quite simple -- anything passed into `res.view` under the key `templates` is turned into a react component. Use it however. Jade content in, React component out. `<Meta />` corresponds to `meta.jade` in my local app templates folder. Its just an example, commonly seen throughout force. Its not actually a formal `<meta>` html tag, but i can see how that could confuse. I've been in force land and am trying to keep things 1-1 and as feature and complexity-free as possible as i build this out.

Comment (damassi) [inline desktop/apps/auctions2/templates/index.jsx]: I want to reiterate that the point of all this is to clear a path for migrating forward, and nothing more. I want to make things _forward compatible_ and then, once that's in place, start thinking about backwards compatibility (as @craigspaeth mentioned in the first PR) which will then quite possibly fit into the work being done over in reaction-force. The two systems are largely incompatible; this makes them compatible (up until a point), but the main blocker was the layout structure for force in general.

Comment (acjay) [inline desktop/apps/auctions2/templates/index.jsx]: Okay, yeah, that confirms my understanding. Which is a very fresh understanding, to be fair, because I didn't actually know how it worked _before_ until I read your original PR :)

I think what I'm suggesting is a somewhat different idea for how apps control those `block`s. Looking at the old `main_layout` Ezel component, it seems like there are 4 blocks it leaves up to the individual subapp to fill or append: `locals`, `head`, `body`, and `scripts`. My understanding is that the goal here is to allow these to be filled _without_ using Pug templates in the apps.

If I'm on the right track there, my idea is that instead of embedding Pug into React, it might make sense to embed React into Pug. I think it may be possible to get around the need to hand the parts of `main_layout` to the `app` code.

It's quite likely that since you're actually writing the code, that there's nuance to this that you see that I don't. And I'm a total blank slate on how reaction-force fits into this. I'll try to find some time to play with it on my own to see if I can PoC what it's my mind.

Comment (damassi) [inline desktop/apps/auctions2/templates/index.jsx]: Ok, that would be great, and makes sense and is indeed the issue. In the meantime would you mind reviewing the rest of the PR? I would like to continue moving forward into auctions2. This is mainly a quick cleanup around the previous one and if we'd like to explore your approach it would take place in a different PR.
